lib/OrdinaryDiffEqCore/src/misc_utils.jl

```diff
@@ -130,3 +130,36 @@ function get_differential_vars(f, u)
 end
 
 isnewton(::Any) = false
+
+
+function _bool_to_ADType(::Val{true}, chunksize, diff_type)
+    Base.depwarn("Using a `Bool` for keyword argument `autodiff` is deprecated. Please use an `ADType` specifier.", :_bool_to_ADType)
+    chunksize = SciMLBase._unwrap_val(chunksize) == 0 ? nothing : SciMLBase._unwrap_val(chunksize)
+    typeof(AutoForwardDiff(chunksize = chunksize))
```

Comment:
why return the type and not the ADTypes object itself?

Reply:
Yes that's the same question I have above in https://github.com/SciML/OrdinaryDiffEq.jl/pull/2508#discussion_r1844930312. Why is the type ever used and why not just use the ADTypes object everywhere? I think there's two clear major changes to make here:

1. Do not accept types, i.e. `autodiff=AutoForwardDiff` shouldn't be an acceptable choice, just `autodiff=AutoForwardDiff()`
2. Never change instances to types internally. There's no need to.
3. Delete all of the code doing weird stuff on types. If you always have instances you never have to do that and you never have to guess any runtime elements.

If ADTypes stays as objects rather than their types, all of these problems go away. Keep life simple.

Reply:
Yeah, I was trying to make this work with as little changes possible, and I didn't think that the ADType objects could be a type parameter. 
But it looks like the algorithms will need an `autodiff` field anyway, so might as well do that now?

Accepting types thing was something weird I had to do because of `remake` .